test/src/Makefile.am

```diff
@@ -1,9 +1,17 @@
+AUTOMAKE_OPTIONS        = subdir-objects
+
 # Add source of new files to be in tarball here
-EXTRA_DIST              = unittest.h xht.c
+EXTRA_DIST              = unittest.h xht.c addr_test.c
 
 # Add new test programs here
-noinst_PROGRAMS         = xht #new-test
+noinst_PROGRAMS         = xht addr
+
+addr_SOURCES            = addr_test.c
+
 LIBS                   += ../../libmdnsd/libmdnsd.la $(cmocka_LIBS)
 
+addr_LDADD              = ../../src/addr.o $(LIBOBJS)
+addr_LDFLAGS            = $(AM_LDFLAGS) -Wl,--wrap=getifaddrs -Wl,--wrap=freeifaddrs
+
 TESTS                   = xht
-# TESTS                  += new-test
+TESTS                  += addr
```

Comment:
Same here, `addr_test` 

(I have no objection to naming the tests differently from the lib files.)